app/controllers/boxes_controller.rb

```diff
@@ -96,12 +104,46 @@ def load_box
     @box = Box.where(institution: @navigation_context.institution).find(params.fetch(:id))
   end
 
+  def load_box_print
+    begin
+      @box = Box.where(institution: @navigation_context.institution).find(params.fetch(:id))
+    rescue ActiveRecord::RecordNotFound
+      @transfer_package = TransferPackage.where("id = ? and ( sender_institution_id = ? or receiver_institution_id = ? )", params.fetch(:transfer_package), @navigation_context.institution['id'], @navigation_context.institution['id']).first
+      @box = Box
+        .joins(box_transfers: :transfer_package)
+        .where( transfer_packages: { id: @transfer_package.id })
+        .find(params.fetch(:id))
+    end
+  end
+
   def load_box_samples
     samples = @box.samples.preload(:batch, :sample_identifiers)
-    samples = samples.scrambled if @box.blinded?
     SamplePresenter.map(samples, request.format)
   end
 
+  def load_box_samples_print
+    samples = @box.samples.preload(:batch, :sample_identifiers)
+    if is_sender or !@box.blinded
+      samples = samples.sort_by{|sample| [sample.batch_number, sample.concentration, sample.replicate ]}
+    else
+      samples = samples.scrambled #this sorts by uuid
+    end
+  end
+
+  def is_sender
+    return @box
+      .box_transfers
+      .joins(:transfer_package)
+      .where( transfer_packages: { id: params.fetch(:transfer_package), sender_institution_id: @navigation_context.institution['id'] } )
+      .present?
+  end
+
+  def floating_box_auth_exception
+    if @box.institution != nil and !BoxTransfer.joins(:transfer_package, :box).where( transfer_packages: { id: params.fetch(:transfer_package), sender_institution_id: @navigation_context.institution['id'] }, box_id: params[:id] ).present?
+      return unless authorize_resource(@box, READ_BOX)
+    end
```

Comment:
There is no `else` to authorize that the user actually has the READ_BOX for either the sender or receiver institution. We can probably do it by reproducing what `authorize_resource` is doing under the hood (I know @straight-shoota did).

Don't worry, I'll deal with that case before merging.